### PR TITLE
Lp 236 general partner choice page

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link

https://companieshouse.atlassian.net/browse/LP-236

### Change description

Page template, routing - back and forward and html validation for the general partner choice page.
Not displaying the partnership name.

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/views/general-partner-choice.njk [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{% extends "layout.njk" %}
+
+{% block beforeContent %}
```
[the remark this answers is not in the document]
Comment (mattch1) [reply src/views/general-partner-choice.njk:3]: Removed unused backlink

File: src/presentation/test/integration/registration/general-partner-choice.test.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  it("should load the general partner choice page with English text", async () => {
+    setLocalesEnabled(true);
+    const res = await request(app).get(GENERAL_PARTNER_CHOICE_URL + "?lang=en");
+
+    expect(res.status).toBe(200);
+    expect(res.text).toContain(enTranslationText.generalPartnerChoicePage.title);
+  });
+
+  it("should redirect to next page when choice is selected", async () => {
+    const selectedType = "Person";
```
[the remark this answers is not in the document]
Comment (mattch1) [reply src/presentation/test/integration/registration/general-partner-choice.test.ts]: reset this value.

**Review by markpit — APPROVED** [review]

**Review by lduranteau — APPROVED** [review]

**Pull request merged**